andela/matterhorn-cfh
# chore: upgrade codebase from es5 to es6

#### What does this PR do?
This PR will upgrade codebase from ES5 to ES6

#### Description of Task to be completed?
N/A

#### How should this be manually tested?
- Clone repo or `git pull`
- Ensure `MONGOHQ_URL` is set as environment variable in a `.env` file
- Run `npm run build:dev` to transpile codebase. Gulp automatically watches the file for changes.
- In another terminal, run `npm run start:dev` to start the development server. Nodemon automatically watches the transpiled files for changes.
- If `PORT` is not specified as an environment variable, you can access the frontend at `http://localhost:3000`

#### Any background context you want to provide?
- The previous codebase uses `cyclic dependencies` for importing modules so the `server.js` file was split into two; `app.js` for exporting express for use in `config/(express|passport|routes).js` and `server.js` for running the server.
- The `gulpfile` has been refactored to use es6 syntax and it is more compact than earlier version.
- To run the app in production, run `npm run build` first and then `npm start`.

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
<img width="1021" alt="screen shot 2017-09-21 at 1 45 46 pm" src="https://user-images.githubusercontent.com/20769821/30696371-4d74d86e-9ed3-11e7-8f4b-db67f69d43e5.png">

#### Questions:
N/A
